src/SerializerFactory.php

```diff
@@ -51,17 +51,28 @@ class SerializerFactory {
 	private $dataValueSerializer;
 
 	/**
+	 * @var ExtraValuesAssigner
+	 */
+	private $extraValuesAssigner;
+
+	/**
 	 * @param Serializer $dataValueSerializer serializer for DataValue objects
+	 * @param ExtraValuesAssigner $extraValuesAssigner
 	 * @param int $options set multiple with bitwise or
 	 *
 	 * @throws InvalidArgumentException
 	 */
-	public function __construct( Serializer $dataValueSerializer, $options = 0 ) {
+	public function __construct(
+		Serializer $dataValueSerializer,
+		ExtraValuesAssigner $extraValuesAssigner,
+		$options = 0
```

Comment:
We should make that an optional parameter and by that avoid the breaking change.